metadata/src/main/java/org/apache/kafka/controller/ReplicationControlManager.java

```diff
@@ -605,6 +605,16 @@ ControllerResult<CreateTopicsResponseData> createTopics(
         Map<ConfigResource, Map<String, Entry<OpType, String>>> configChanges =
             computeConfigChanges(topicErrors, request.topics());
 
+        int totalPartitions = 0;
+        for (CreatableTopic topic: request.topics()) {
+            if (topic.numPartitions() > 0) {
+                totalPartitions += topic.numPartitions();
+            }
+        }
+        if (totalPartitions > 10_000) {
```

Comment:
can we use a constant for this (static final, etc. int).

That would also allow us to lock it to the batch size constant

Reply:
Done (I think) - is there a specific constant you think we should make it a `multiple` of?